### PR TITLE
Bitmaker-845: Add support for job args

### PR DESCRIPTION
**Pull request by rodp63**
#### Ticket url(s):
- https://tasks.bitmaker.dev/issues/845

#### Description:
_Add support for job args._

#### Checklist:
- [x] Check that only one commit is added in the PR with the correct format
- [X] Rebase my branch

### CODE REVIEW

File: bm_scrapy/__main__.py [hunk 1]
```diff
@@ -28,15 +28,15 @@ def run_code(args, commands_module=None):
 
 
 def describe_project():
-    # bm-describe-project command
+    """bm-describe-project command."""
```
Comment (renato262) [inline bm_scrapy/__main__.py]: Docstring needs to be a sentence like `Describe project.`. A simple and direct sentence in present simple.

File: bm_scrapy/__main__.py [hunk 2]
```diff
@@ -54,13 +54,15 @@ def setup_and_launch():
 
     
 def main():
-    # bm-crawl command
+    """bm-crawl command."""
+    """
     try:
```
Comment (renato262) [inline bm_scrapy/__main__.py]: Is not a good practice to comment code, please fix and fix the docstring also

File: tests/test_main.py [hunk 3]
```diff
@@ -62,10 +62,12 @@ def test_describe_project(mock_run_code, mock_setup_scrapy_conf):
     assert run_args[1] == 'bm_scrapy.commands'
     
 
+"""
```
Comment (renato262) [inline tests/test_main.py]: same here

File: tests/test_utils.py [hunk 4]
```diff
@@ -5,4 +5,4 @@
 
 def test_parse_time():
     great_date = datetime(1985, 8, 14, 7, 10, 1, 123456)
-    assert parse_time(great_date) == '14/08/1985, 07:10:01.123'
+    assert parse_time(great_date) == '14/08/1985 07:10:01.123'
```
Comment (renato262) [inline tests/test_utils.py:8]: why did we erase the `,`?

**Review by renato262 — CHANGES_REQUESTED** [review]

**Comment by renato262** [comment]
@rodp63  What are we passing in `arg1` and `arg2`?

File: bm_scrapy/__main__.py (hunk 1 above)
Comment (rodp63) [reply bm_scrapy/__main__.py]: Done

File: tests/test_utils.py (hunk 4 above)
Comment (rodp63) [reply tests/test_utils.py:8]: We should use a standard format specifier. https://docs.microsoft.com/en-us/dotnet/standard/base-types/standard-date-and-time-format-strings#GeneralDateLongTime
> The "G" standard format specifier represents a combination of the short date ("d") and long time ("T") patterns, separated by a space

File: bm_scrapy/__main__.py (hunk 2 above)
Comment (rodp63) [reply bm_scrapy/__main__.py]: Good advice, I'll change it.

File: tests/test_main.py (hunk 3 above)
Comment (rodp63) [reply tests/test_main.py]: Done

**Comment by rodp63** [comment]
`args` is a dictionary within the env variable `JOB_INFO` containing the args of the spider job. e.g. we should use `JOB_INFO={"spider": "metro", "key":"pid/sid/jid", "args":{"job_type":"products", "urls":"https://www.metro.pe/product-x"}}` for launching a market crawler.

File: tests/test_main.py [hunk 5]
```diff
@@ -60,12 +60,9 @@ def test_describe_project(mock_run_code, mock_setup_scrapy_conf):
     run_args = mock_run_code.call_args[0]
     assert run_args[0] == expected_args
     assert run_args[1] == 'bm_scrapy.commands'
-    
 
-@mock.patch('bm_scrapy.writer.pipe_writer')
+
 @mock.patch('bm_scrapy.__main__.setup_and_launch')
-def test_main( mock_setup_and_launch, mock_pipe_writer):
+def test_main( mock_setup_and_launch):
```
Comment (renato262) [inline tests/test_main.py]: @rodp63 Remove the unnecessary blank space between `(` and `mock_setup_and_launch` and we are ready to merge

**Review by renato262 — CHANGES_REQUESTED** [review]

File: tests/test_main.py (hunk 5 above)
Comment (rodp63) [reply tests/test_main.py]: Done

**Review by renato262 — APPROVED** [review]

**Pull request merged**